parties/googleclient.go

```diff
@@ -87,88 +93,124 @@ func (g *GoogleOp) RequestTokens(cicHash string, cb TokenCallback) error {
 	logrus.Info("press ctrl+c to stop")
 	earl := fmt.Sprintf("http://localhost:%s/login", g.RedirURIPort)
 	openUrl(earl)
-	logrus.Fatal(g.server.ListenAndServe())
 
-	return nil
-}
+	go func() {
+		err := g.server.ListenAndServe()
+		if err != nil && err != http.ErrServerClosed {
+			logrus.Fatal(err)
+		}
+	}()
 
-func (g *GoogleOp) VerifyPKToken(pktJSON []byte, cosPk *ecdsa.PublicKey) error {
+	defer g.server.Shutdown(context.TODO())
 
-	pkt, err := pktoken.FromJSON(pktJSON)
-	if err != nil {
-		logrus.Fatalf("Error parsing PK Token: %s", err.Error())
-		return err
+	select {
+	case err := <-chErr:
+		return nil, err
+	case token := <-ch:
+		return token, nil
 	}
+}
 
-	nonce, err := pkt.GetNonce()
+func (g *GoogleOp) VerifyPKToken(pktJSON []byte, cosPk *ecdsa.PublicKey) (map[string]any, error) {
+	pkt, err := pktoken.FromJSON(pktJSON)
 	if err != nil {
 		logrus.Fatalf("Error parsing PK Token: %s", err.Error())
-		return err
+		return nil, err
 	}
 
-	options := []rp.Option{
-		rp.WithVerifierOpts(rp.WithIssuedAtOffset(5*time.Second), rp.WithNonce(func(ctx context.Context) string { return string(nonce) })),
-	}
-
-	googleRP, err := rp.NewRelyingPartyOIDC(
-		g.Issuer, g.ClientID, g.ClientSecret, g.RedirectURI, g.Scopes,
-		options...)
-
+	cicphJSON, err := util.Base64DecodeForJWT(pkt.CicPH)
```

Comment:
Can you think of any reason we shouldn't change the spec to hash the base64 encoding of the CIC? That way a verifier would never need to decode a CIC protected header that wasn't committed to in nonce. It would prevent someone from messing with the [unused bits in a base64 encoding.](https://developer.fastly.com/reference/vcl/functions/cryptographic/digest-base64-decode/#:~:text=Unused%20bits&text=These%20ignored%20bits%20could%20have,first%20two%20bits%20of%20A%20.) 

This also fits closer with how JWS computes the signature:

> Compute the JWS Signature in the manner defined for the
       particular algorithm being used over the JWS Signing Input
       ASCII(BASE64URL(UTF8(JWS Protected Header)) || '.' ||
       BASE64URL(JWS Payload)). 
       
https://www.rfc-editor.org/rfc/rfc7515.html#section-5.1

Reply:
I think this is a great idea! As an aside, I guess we should we have a repo to host the spec in the openpubkey org?

Reply:
Yes, I think so